mangaki/mangaki/wrappers/anilist.py

```diff
@@ -247,8 +289,19 @@ def _request(self,
             query_params = {}
 
         r = self._session.get(urljoin(self.BASE_URL, datapage.format(**params)), params=query_params)
+        data = r.json()
+
+        if r.status_code == 200 or r.status_code == 404:
```

Comment:
You might want to inline those nested if as `and` conditions:

``python
if r.status_code in (200, 404) and not isinstance(data, list) and data.get('error'):
     if is_no_results(data.get('error')):
         return None
     else:
        raise AniListError(data)
```

Something along these lines may be better?
Also, we could skip the status_code check, we want to check errors *for all responses*.